File: ALifeCommon/src/main/java/com/cristiano/alife/instructions/transfers/SaveToRegisterPositionInstruction.java
package com.cristiano.alife.instructions.transfers;

import com.cristiano.alife.consts.ALifeConsts;
import com.cristiano.alife.world.BaseInstruction;
import com.cristiano.alife.world.IOrganismo;
import com.cristiano.alife.world.IWorld;

/*
 * STAX: (to) <- from
 * */
public class SaveToRegisterPositionInstruction extends BaseInstruction {
	public SaveToRegisterPositionInstruction(IWorld mundo) {
		super(mundo);
	}

	@Override
	public void executa(IOrganismo o) {
		int inst = o.getReg(getByteOrganismo(o, 1));
		inst = mundo.getMutation().mutateInstruction(inst,o);
		/*
		 * if (!mundo.isValidInstruction(inst)){ o.fatalError(); return; }
		 */

		int toPos = o.getReg(getByteOrganismo(o, 2));
		o.setMemory(toPos, inst);
		o.wroteFlag();
	}

	@Override
	public int getStep() {
		return 3;
	}

	@Override
	public String getDescription(IOrganismo o, int ip) {
		return comment(name + " " + ALifeConsts.getLetter(o.getMemory(ip + 1)) + "," + ALifeConsts.getLetter(o.getMemory(ip + 2)),
				"(" + ALifeConsts.getLetter(o.getMemory(ip + 2)) + ") <- " + ALifeConsts.getLetter(o.getMemory(ip + 1)) + "");
	}

}

File: netAvida/netAvida.Backend/instructions/impl/transfers/SaveToRegisterPositionInstruction.cs
using netAvida.backend;
using netAvida.backend.interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace netAvida.Backend.instructions.impl
{
    public class SaveToRegisterPositionInstruction : BaseInstruction
    {
        public SaveToRegisterPositionInstruction(IWorld mundo): base(mundo)
        {
        }

    public override void executa(IOrganismo o)
        {
            int a = o.getReg(getByteOrganismo(o, 1));
            int b = o.getReg(getByteOrganismo(o, 2));
            o.setReg(getByteOrganismo(o, 2), a);
            o.setReg(getByteOrganismo(o, 1), b);
        }

    public override int getStep()
        {
            return 3;
        }

    public override String getDescription(IOrganismo o, int ip)
        {
            return comment(name + " " + ALifeConsts.getLetter(o.getMemory(ip + 1)) + "," + ALifeConsts.getLetter(o.getMemory(ip + 2)), ALifeConsts.getLetter(o.getMemory(ip + 1)) + " <-> " + ALifeConsts.getLetter(o.getMemory(ip + 2)) + "");
        }
    }
}
